sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/parquet/GpuParquetReader.java

```diff
@@ -139,8 +139,8 @@ public org.apache.iceberg.io.CloseableIterator<ColumnarBatch> iterator() {
           new Path(input.location()), clippedBlocks, fileReadSchema, caseSensitive,
           partReaderSparkSchema, debugDumpPrefix, debugDumpAlways,
           maxBatchSizeRows, maxBatchSizeBytes, targetBatchSizeBytes, useChunkedReader, metrics,
-          true, // isCorrectedInt96RebaseMode
-          true, // isCorrectedRebaseMode
+          "CORRECTED", // dateRebaseMode
```

Comment:
Not a fan of using and matching on strings for what's essentially an enum.  It's easy to typo and the compiler won't help catch it.  If we made these symbolic then it can.  Can use a sealed trait and case objects to do that, see GpuBuildSide, RegexMode, FileFormatType, FileFormatOp, etc.

Reply:
I tried to use the original enum from Spark. Scala code can compile fine but this Java code doesn't. It says `"Symbol not found"` for `LegacyBehaviorPolicy`. Do you have any idea?

Reply:
The import should have failed, there is no Java class named LegacyBehaviorPolicy because there's no Scala LegacyBehaviorPolicy class, only an object.  Scala objects are name-mangled when translated to Java classes.  The Scala object's class has a `$` appended, and it has a static member that is an instance of the class named `MODULE$`.  Also most things that look like fields in Scala are actually implemented as Java methods.  For example, to access the EXCEPTION enum value, you'll need to something like this in Java:
```java
import org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy$;
[...]
scala.Enumeration.Value mode = ...;
if (mode == LegacyBehaviorPolicy$.MODULE$.EXCEPTION()) {
   ...
```